File: Transport/@stateflows/common/src/events/notifications.request.ts
import { Request } from "./request";
import { NotificationsResponse } from "./notifications.response";
import { StateflowsEvent } from "../decorators/stateflows-event";

@StateflowsEvent("Stateflows.Common.NotificationsRequest, Stateflows.Common")
export class NotificationsRequest extends Request<NotificationsResponse> { }

File: Core/Stateflows.Common/Events/NotificationsRequest.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Stateflows.Common
{
    [NoImplicitInitialization]
    public sealed class NotificationsRequest : IRequest<NotificationsResponse>
    {
        public TimeSpan Period { get; set; }
        
        [JsonProperty(TypeNameHandling = TypeNameHandling.None)]
        public List<string> NotificationNames { get; set; } = new List<string>();
        
        public void Add<TNotification>()
            => NotificationNames.Add(Event<TNotification>.Name);
    }
}
